reposilite-backend/src/main/kotlin/com/reposilite/console/infrastructure/ConsoleSseHandler.kt

```diff
@@ -0,0 +1,112 @@
+package com.reposilite.console.infrastructure
+
+import com.reposilite.ReposiliteJournalist
+import com.reposilite.auth.AuthenticationFacade
+import com.reposilite.auth.api.Credentials
+import com.reposilite.console.api.ExecutionResponse
+import com.reposilite.shared.ErrorResponse
+import com.reposilite.shared.extractFromHeader
+import com.reposilite.shared.unauthorized
+import com.reposilite.token.AccessTokenFacade
+import com.reposilite.token.AccessTokenPermission
+import io.javalin.http.Context
+import io.javalin.http.Header
+import io.javalin.http.sse.SseClient
+import io.javalin.openapi.HttpMethod
+import io.javalin.openapi.OpenApi
+import io.javalin.openapi.OpenApiContent
+import io.javalin.openapi.OpenApiResponse
+import panda.std.Result
+import panda.std.reactive.Reference
+import java.nio.channels.ClosedChannelException
+import java.util.*
+import java.util.function.Consumer
+
+private const val SSE_EVENT_NAME = "log"
+
+data class SseSession(
+    val identifier: String,
+    val subscriberId: Int,
+)
+
+internal class ConsoleSseHandler(
+    private val journalist: ReposiliteJournalist,
+    private val accessTokenFacade: AccessTokenFacade,
+    private val authenticationFacade: AuthenticationFacade,
+    private val forwardedIp: Reference<String>
+) : Consumer<SseClient> {
+
+    internal val users: WeakHashMap<SseClient, SseSession> = WeakHashMap()
+
+    @OpenApi(
+        // TODO: does this need better endpoint name?
+        path = "/api/console/log",
+        methods = [HttpMethod.GET],
+        // TODO: the responses parameter is *technically* not really needed, but would be nice to have
+        responses = [
+            OpenApiResponse(
+                status = "200",
+                // TODO: better description?
+                description = "Continuously sends out the log as data messages",
+                content = [OpenApiContent(from = ExecutionResponse::class)]
+            )
+        ],
+        tags = ["Console"]
+    )
+    override fun accept(sse: SseClient) {
+        sse.keepAlive()
+        sse.onClose {  ->
+            val session = users.remove(sse) ?: return@onClose
+            journalist.logger.info("CLI | ${session.identifier} closed connection")
+            journalist.unsubscribe(session.subscriberId)
+        }
+
+        when (users[sse]) {
+            null ->
```

Comment:
I think it is redundant, each SseClient instance will be treated as a different value, so it'll be always null for a new connection.